datafusion/sql/src/unparser/plan.rs

```diff
@@ -215,7 +163,11 @@ impl Unparser<'_> {
             LogicalPlan::Filter(filter) => {
                 let filter_expr = self.expr_to_sql(&filter.predicate)?;
 
-                select.selection(Some(filter_expr));
+                if let LogicalPlan::Aggregate(_) = filter.input.as_ref(){
```

Comment:
I suspect we will find more test cases where this logic is not sufficient to determine if a filter should be in a HAVING or WHERE clause.